### ISSUE TITLE
Job not launched

### ISSUE THREAD

**Issue by pvanacker** [rundeck/rundeck#1712]
One of our jobs didn't launch and nothing is displayed in the activities tab for this job.

There is a java exception in logs :

2016-02-29 02:00:00,178 [quartzScheduler_Worker-8] ERROR grails.app.jobs.rundeck.quartzjobs.ExecutionJob - Unable to start Job execution: No row with the given identifier exists: [rundeck.Workflow#120]
org.hibernate.ObjectNotFoundException: No row with the given identifier exists: [rundeck.Workflow#120]
	at rundeck.services.ExecutionService$_int_createExecution_closure44.doCall(ExecutionService.groovy:1578)
	at rundeck.services.ExecutionService.int_createExecution(ExecutionService.groovy:1577)
	at rundeck.services.ExecutionService.createExecution(ExecutionService.groovy:1679)
	at rundeck.quartzjobs.ExecutionJob.initialize(ExecutionJob.groovy:265)
	at rundeck.quartzjobs.ExecutionJob.execute_internal(ExecutionJob.groovy:95)
	at rundeck.quartzjobs.ExecutionJob$_execute_closure1.doCall(ExecutionJob.groovy:74)
	at com.codahale.metrics.Timer.time(Timer.java:99)
	at rundeck.quartzjobs.ExecutionJob.execute(ExecutionJob.groovy:73)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)

It usually launches without this problem and I was able to relaunch it manually later.

Rundeck version : 2.6.2
It doesn't seem to be an issue corrected in version 2.6.3

Thanks,

Pierre

**Comment by minhuber** [comment]
I have the very same issue and it's randomly occurring. E.g. in the job: "rcm_schedule" running every 15 minutes, I got yesterday April 3rd: 11, on April 2nd: 5 failures with this message...

**Comment by gschueler** [comment]
what type of database type are you using?

**Comment by pvanacker** [comment]
The default h2 database

**Comment by minhuber** [comment]
Using galera cluster with MariaDB. However as we configured 2 rundeck servers in a cluster, the connections from the rundeck servers are loadbalanced to the galera cluster MariaDB (active - active configuration).
-> The stack trace actually indicates that the error is DB connection related. Changing the loadbalancer configuration monitor scan time-outs from 3 seconds to 5 seconds (and if not responding initiate a failover) resolved this issue.
The current statistics on the test job shows now 549 executions in the last 6 days with 100% success.
In other words, this issue is obsolete for us now.
@pvanacker, I guess it would be helpful to know if the h2 database is on the same server as the rundeck server.

**Comment by pvanacker** [comment]
Indeed, the h2 database is on the same server as the rundeck server.

**Comment by gschueler** [comment]
I don't know how to reproduce this issue for h2. The error indicates it is failing to load the Workflow from the job definition, the only thing I can think of is that the Job was being modified at the time the schedule triggered execution, and a new Workflow definition was created, such that the reference (db row id) to the old workflow was no longer valid at the moment the job was triggered.

@pvanacker was it possible that the Job definition was modified (user editing in the GUI, or definition uploaded via API) at the times that those errors occurred?

**Comment by pvanacker** [comment]
The job was scheduled to launch during the night and it happened multiple times so i don't think the job definition was modified (not many people have access to our rundeck instance)
However we didn't encountered this issue since a month.

I'll keep looking at it, if i find something that can help reproduce it, i'll let you know.

Thank you

**Comment by gschueler** [comment]
closing this. please let me know if you can reproduce it, thanks.
